### ISSUE TITLE
Disable WAL-mode File Format in sqlite for Android 9

### ISSUE THREAD

**Issue by avarelahn** [brody2consult/cordova-sqlite-storage#887]
Hello, I  have a problem with my application in the android pie version.
When executing the select using this.sqlite.create () to my database, I get the error: code: 5
message: "sqlite3_prepare_v2 failure: no such table: products"
Apparently I created the database but not the table, in previous versions my application worked great, but in the Pie version or greater it gives me this problem. I read that Android 9 introduces an activated default functionality called WAL (write-ahead logging) but I don't know how to disable this in my code, someone could help me with this, I would appreciate it.

---

From @brodybits: this is only an issue when the `androidDatabaseProvider: 'system'` setting is used when opening the database. See below for further discussion with the user community.

**Comment by brody2consult** [comment]
A minimal reproducible example is needed: https://stackoverflow.com/help/minimal-reproducible-example

In case you need help with this, commercial support is available from: https://xpbrew.consulting/

**Comment by AhmadAyyaz1993** [comment]
I am getting the same problem. Is there any way? 
Here is the reference related to this issue https://stackoverflow.com/a/53917295

**Comment by AhmadAyyaz1993** [comment]
At the moment I disabled write ahead logging in this method of the plugin 
`void open(File dbfile) throws Exception {
        dbFile = dbfile; // for possible bug workaround
        mydb = SQLiteDatabase.openOrCreateDatabase(dbfile, null);
	mydb.disableWriteAheadLogging();
    }` 
but I am not sure if it will work. Because google suggests to use it in onConfigure of DatabaseHelper class.

**Comment by brody2consult** [comment]
This plugin uses a custom build of SQLite unless you specify otherwise when you open the database. The custom build of SQLite has **never** used WAL mode.

This information is already documented but is evidently not clear enough.

**Comment by AhmadAyyaz1993** [comment]
So that means we don't need to disable WAL mode specifically anywhere in plugin? if we are using 'databaseprovide: system' ?

**Comment by julientype** [comment]
databaseprovide: system   <=== creates a persistent journal database running 2 files or whatever FLAVOR THE os IS RUINING..... WAL 

The use of persistent journal mode provides a noticeable performance improvement on many systems. Of course, the drawback is that the journal files remain on the disk, using disk space and cluttering directories, long after the transaction commits. The only safe way to delete a persistent journal file is to commit a transaction with journaling mode set to DELETE:

PRAGMA journal_mode=DELETE;
BEGIN EXCLUSIVE;
COMMIT;
Beware of deleting persistent journal files by any other means since the journal file might be hot, in which case deleting it will corrupt the corresponding database file.

Basically your not running an OS data system where you need a persistent journal like FOR reboot android THE persistent journal  IS CORRUPTED 

an android app is just a single user..........

**Comment by brody2consult** [comment]
Thanks to all for the information. I have just added some info to the description and pinned this issue for the benefit of the user community.

**Comment by julientype** [comment]
Exclusive Access Mode
SQLite version 3.3.14 adds the concept of "Exclusive Access Mode". In exclusive access mode, SQLite retains the exclusive database lock at the conclusion of each transaction. This prevents other processes from accessing the database, but in many deployments only a single process is using a database so this is not a serious problem. The advantage of exclusive access mode is that disk I/O can be reduced in three ways:

It is not necessary to increment the change counter in the database header for transactions after the first transaction. This will often save a write of page one to both the rollback journal and the main database file.

No other processes can change the database so there is never a need to check the change counter and clear the user-space cache at the beginning of a transaction.

Each transaction can be committed by overwriting the rollback journal header with zeros rather than deleting the journal file. This avoids having to modify the directory entry for the journal file and it avoids having to deallocate disk sectors associated with the journal. Furthermore, the next transaction will overwrite existing journal file content rather than append new content and on most systems overwriting is much faster than appending.

The third optimization, zeroing the journal file header rather than deleting the rollback journal file, does not depend on holding an exclusive lock at all times. This optimization can be set independently of exclusive lock mode using the journal_mode pragma 

===== YOUR SIMPLY RUNNING SINGLE USER APP   RIGHT ======
ON START SET DB=OPEN 
  DB.CLOSE APP CLOSE
ONLY ONE CONNECTION 
so you just run  in  exclusive mode...... if your on a web server you need persistent journal where admin  edits and everyone reads from ....... 2 files 
so it depends on what..... where one function will create the execution order your in exclusive mode you need a progress bar..... getting data... its  an app it serves just one client yet if you want to run record session with record locking you can.... file server open connection but this just android
